File: 从程序员到架构师/1-hw/old/homework/2th-第二周/3-第三天/src/main/java/com/gyzh/zain/vertx/web/Bootstrap.java
package com.gyzh.zain.vertx.web;

import io.vertx.core.AbstractVerticle;
import io.vertx.core.Future;
import io.vertx.core.Handler;
import io.vertx.core.http.HttpServer;
import io.vertx.core.http.HttpServerRequest;
import io.vertx.core.http.HttpServerResponse;
import io.vertx.ext.web.Router;
import io.vertx.ext.web.sstore.LocalSessionStore;
import io.vertx.ext.web.handler.*;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.File;

public class Bootstrap extends AbstractVerticle {

    Logger log = LoggerFactory.getLogger(this.getClass());

    @Override
    public void start(Future<Void> startFuture) throws Exception {
        HttpServer server = vertx.createHttpServer();
        Router router = Router.router(vertx);

        router.route().handler(CookieHandler.create());
        router.route().handler(BodyHandler.create());
        router.route().handler(SessionHandler.create(LocalSessionStore.create(vertx)));


        // Handles the actual login
        router.route("/loginhandler").handler(routingContext -> {
            HttpServerResponse response = routingContext.response();
            response.putHeader("Content-type", "text/html;charset=UTF-8");
            response.end("登陆成功！");
        });

        // 注册登出
        router.route("/logout").handler(context -> {
            context.response().putHeader("location", "/").setStatusCode(302).end();
        });

        // 注册静态文件访问，默认在resources下
        router.route().handler(StaticHandler.create());

        server.requestHandler(router::accept).listen(8881);
    }
}

File: 从程序员到架构师/课程1/1-hw/old/homework/2th-第二周/3-第三天/src/main/kotlin/com/gyzh/zain/vertx/web/Bootstrap.kt
package com.gyzh.zain.vertx.web

import io.vertx.core.Vertx
import io.vertx.ext.web.Router
import com.gyzh.zain.vertx.web.handler.*
import java.util.*

val vertx = Vertx.vertx()

fun main (args: Array<String>) {

    var server = vertx.createHttpServer()
    var router = createRouter()

    server.requestHandler({ router.accept(it) }).listen(8081)
}

/**
 * 路由创建器
 */
private fun createRouter() = Router.router(vertx).apply {
    get("/").handler(rootHandler)
    get("/count").handler(countHandler)
    get("/count/1").handler({ req ->
        req.fail(403)
    })
    get("/*").failureHandler({ frc ->
        var statusCode = frc.statusCode() // 获取异常码
        var response = frc.response() // 获取响应

        var errText = arrayOf("不好意思，您找错了"
                , "您在转转吧"
                , "还是没找对"
                , "这个地方不存在")
        var rNo = Random().nextInt(10)
        System.out.println("no: " + rNo)
        var no = rNo % 4
        var text = errText[no]
        System.out.println("text: " + text)
        response.putHeader("Content-type", "text/html;charset=UTF-8");
        response.setStatusCode(statusCode).end(text)
    })
}
